packages/drivers/odsp-driver/src/odspDriverUrlResolver2.ts

```diff
@@ -62,12 +62,13 @@ export class OdspDriverUrlResolver2 implements IUrlResolver {
      */
     public async resolve(request: IRequest): Promise<IOdspResolvedUrl> {
         const requestToBeResolved = { headers: request.headers, url: request.url };
-        let sharingLink: string | undefined;
+        let sharingLinkP: Promise<string> | undefined;
+        const isSharingLink = requestToBeResolved.headers?.[SharingLinkHeader.isSharingLink];
```

Comment:
I would make this an optional param (from the caller point of view) and default to false if it's undefined. That way if you want to add more headers in the future it's not just "all or nothing" in terms of headers.

Reply:
Supplying SharingLinkHeader.isSharingLink is optional. And if it not supplied we treat it as false.
Is it whjat you are saying?

Reply:
Or are you saying to make it optional parameter in resolve and remove it from header?

Reply:
Ahh maybe I just don't understand how the header construct you have works, it seemed like it was required if you were using the header object. This should be good then :)

Reply:
yeah, header is just optional.